### ISSUE TITLE
Strange behaviour in overlapping bounding boxes

### ISSUE THREAD

**Issue by AntMorais** [XinzeLee/PolygonObjectDetection#17]
@XinzeLee I have an issue when there are two adjacent or overlapping objects that I want to detect. I created a diagram to give an example.

![PolygonProblem](https://user-images.githubusercontent.com/26347971/153185950-d37eac89-b071-4d04-bd4c-507e3dfe9cc5.png)

The objects I am trying to detect have a rectangular shape and are represented in the image by black rectangles with grey outlines. The bounding boxes predicted by  the model are in red. 

As you can see, box 1 and 3 are correct and box 2 is incorrect. Increasing the confidence or decreasing the IoU threshold cause only box 2 to be visible. 

This happens in almost every case where 2 or more objects are close together.

Any idea on the root of the problem?

**Comment by XinzeLee** [comment]
From what you have mentioned, clearly the model is giving the box 2 a higher confidence than the other box 1 and 3, explaining why increasing confidence threshold or reduce IoU threshold won't help.

Regarding this issue, I have several possible solutions for your:
1. Check your training process carefully. Is the training process already overfitting or still underfitting. If it is overfitting, try to reduce the training epochs. If it is underfitting, then try to allow the model to train more.
2. If the training process is not the reason, then maybe assign a higher weight to the object loss (lobj). It might help with the wrong boxing problem.

**Comment by AntMorais** [comment]
The model was in fact overfitting. I improved the problem by adding more data and making sure I don't overfit.
Thank you for you answer and great work on this repo!
